libraries/chain/db_maint.cpp

```diff
@@ -800,6 +800,70 @@ void update_and_match_call_orders( database& db )
    wlog( "Done updating all call orders for hardfork core-343 at block ${n}", ("n",db.head_block_num()) );
 }
 
+/******
+ * @brief cleanup feeds with wrong assets
+ *
+ * Prior to hardfork 868, switching a bitasset's shorting asset would not reset its
+ * feeds. This method will run at the hardfork time, and erase (or nullify) feeds
+ * that have incorrect backing assets.
+ *
+ * @param db the database
+ */
+void cleanup_invalid_feeds_hf_868( database& db )
+{
+   // for each market issued asset
+   const auto& asset_idx = db.get_index_type<asset_index>().indices().get<by_type>();
+   for(auto asset_itr = asset_idx.lower_bound(true); asset_itr != asset_idx.end(); ++asset_itr)
+   {
+      const auto& asset = *asset_itr;
+      // Incorrect witness & committee feeds can simply be removed.
+      // For non-witness-fed and non-committee-fed assets, set incorrect
+      // feeds to price(), since we can't simply remove them. For more information:
+      // https://github.com/bitshares/bitshares-core/pull/832#issuecomment-384112633
+      bool is_witness_or_committee_fed = false;
+      if ( asset.options.flags & (witness_fed_asset | committee_fed_asset) )
+         is_witness_or_committee_fed = true;
+
+      // for each feed
+      const asset_bitasset_data_object& bitasset_data = asset.bitasset_data(db);
+      bool feeds_changed = false;
+      auto itr = bitasset_data.feeds.begin();
+      while (itr != bitasset_data.feeds.end())
+      {
+         // If the feed is invalid
+         if ( (*itr).second.second.settlement_price.quote.asset_id != bitasset_data.options.short_backing_asset
+               && (is_witness_or_committee_fed || (*itr).second.second.settlement_price != price() ) ) {
+            feeds_changed = true;
+            db.modify(bitasset_data, [&itr, is_witness_or_committee_fed](asset_bitasset_data_object& obj) {
+               if (is_witness_or_committee_fed)
+               {
+                  // erase the invalid feed
+                  itr = obj.feeds.erase(itr);
+               }
+               else
+               {
+                  // nullify the invalid feed
+                  (*obj.feeds.find( (*itr).first )).second.second.settlement_price = price();
+                  ++itr;
+               }
+            });
+         }
+         else
+         {
+            // Feed is valid. Skip it.
+            ++itr;
+         }
+      }
+      // if the feeds were modified, update the median feed
+      if (feeds_changed) {
+         db.modify(bitasset_data, [&db](asset_bitasset_data_object &obj) {
+            obj.update_median_feeds( db.head_block_time() );
```

Comment:
Note: likely HF 868 and HF 343 will occur at same time.
* If it's the case, we'll update and check call orders in HF 343 with new rule, so no need to call `check_call_orders()` here.
* Otherwise, if the median feed changed after `update_median_feeds()`, need to call `check_call_orders()` .